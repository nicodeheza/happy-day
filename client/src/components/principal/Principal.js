import React, { useState } from 'react'
import Card from './cards/Card';
import Header from './header/Header';
import Toolbar from './toolbar/Toolbar'
import './principal.css';
import Calendar from './calendar/Calendar';




export default function Principal() {
    const [showCard, setShowCard]= useState('none');
    return (
        <div>
            <Header/>

            <div className="tool-card">

            <Card showCard={showCard} setShowCard={setShowCard}/>
        
            <Toolbar setShowCard={setShowCard}/>
            </div>

            <Calendar/>
        </div>
    )
}
