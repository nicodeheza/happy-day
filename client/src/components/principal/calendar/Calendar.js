import React, { useContext, useEffect, useState } from 'react';
import "./calendar.css"
import {principalContext} from '../Principal';

const monthNames=['January', 'February', 'March', 'April', 'May', 'June', 'July',
 'August', 'September', 'October', 'November', 'December'];

 const monthColors=['#216b8e', '#27a3a3','#149e84', '#00ae4d', '#69d503', '#d6df24', '#edc01c', '#faa818',
  '#f19616', '#fd6126', '#9458a9', '#6258a2'];
  let lastId;


export default function Calendar({setShowCard, showCard}) {
    const[calendarData , setCalendarData]= useState({});
    const {updateCalendar, setUpdateCalendar, setEdit, searchFilters}= useContext(principalContext);
    const [searchResults, setSearchResults]= useState({});
    const [loadingData, setLoadingData]= useState(true);

    const fetchData=()=>{
        //setLoadingData(true)
        fetch('http://localhost:4000/api/events',{
            method:'GET',
            headers:{"Content-type": "application/json; charset=UTF-8"},
            credentials: 'include'
        })
        .then(res=> res.json())
        .then(data=>{
    
          //console.log("fetch calendar.js ");
          setCalendarData(data);
          setLoadingData(false);
          
        })
        .catch(err => console.log(err));
    }

    useEffect(()=>{
       fetchData();
    },[]);

    useEffect(()=>{
        if(updateCalendar){
        fetchData();
        setUpdateCalendar(false);
        }
     },[updateCalendar, setUpdateCalendar]);

    useEffect(()=>{
        
        //console.log(searchFilters);
        let filterData={};
        let fromMonth;
        let fromDay;
        let toMonth;
        let toDay;

        if(searchFilters.from){
            const from= searchFilters.from.split('/');
             fromMonth= parseInt(from[0]);
             fromDay= parseInt(from[1]);
        }else{
            fromMonth= 0;
            fromDay= 0;
        }
        if(searchFilters.to){
            const to= searchFilters.to.split('/');
             toMonth= parseInt(to[0]);
             toDay= parseInt(to[1]);
        }else{
            toMonth= 99;
            toDay= 99;
        }

        let i=0;
        let lastMonth;
        for(const month in calendarData){
            if(month >= fromMonth && month <= toMonth){
                i++;
                lastMonth=month;

                if(i===1){
                    filterData[month]= {};
                    for(const day in calendarData[month]){

                        if(day >= fromDay){
                            filterData[month][day]= [...calendarData[month][day]];
                        }
                    }
                }else{
                    filterData[month]={...calendarData[month]};
                    
                }
                

            }

        }

        for( const day in filterData[lastMonth]){
            if(day > toDay){
                delete filterData[lastMonth][day];
            }
        }

        if(searchFilters.name){
        for(const month in filterData){
            for(const day in filterData[month]){
                 //console.log( filterData);
               const nameFilter= filterData[month][day].filter(data=>{
                   let regExp= new RegExp(searchFilters.name.toLowerCase(), 'ig');
                   return regExp.test(data.personName);
                });
                if(nameFilter.length === 0){
                   delete filterData[month][day];
                }else{
                    filterData[month][day]= nameFilter;
                }
            }

            if(Object.keys(filterData[month]).length === 0){
                delete filterData[month];
            }
        }
     }   

     if(searchFilters.type !== "any" && searchFilters.type){

        for(const month in filterData){
            for(const day in filterData[month]){
               const typeFilter= filterData[month][day].filter(data=>{
                  if(data.type === searchFilters.type){
                      return true;
                  }else{
                      return false;
                  }
                });
                if(typeFilter.length === 0){
                   delete filterData[month][day];
                }else{
                    filterData[month][day]= typeFilter;
                }
            }

            if(Object.keys(filterData[month]).length === 0){
                delete filterData[month];
            }
        }

     }


        //console.log(filterData);
        setSearchResults(filterData);

    },[searchFilters, calendarData]);
    

    const getAge=(date)=>{
        const ageInMs= Date.now() - new Date(date).getTime();
        const ageInYears= Math.floor(ageInMs / (1000*60*60*24*365));
        if(ageInYears > 0){
            return `(${ageInYears} eyars)`;
        }else{
            return '';
        }
    }


    const eventEdit=(id, event)=>{
        //ele.className="event-container selected"
        if(lastId){
            lastId.style.backgroundColor = "transparent";
        }
       const ele= document.getElementById(id);
       ele.style.backgroundColor = "white";
       lastId= document.getElementById(id);

       setShowCard('edit');
       setEdit(event);

    }
    useEffect(()=>{
        if(showCard === "none" && lastId){
            lastId.style.backgroundColor = "transparent";
           }
    },[showCard])


    return (
        <div className="calendar">

           {
               Object.keys(calendarData).length > 0 && !loadingData ?
               Object.keys(searchResults? searchResults : calendarData).map((month, i)=>{
                   return(
                    <div className="month-container" id={"month"+month} key={i}>
                        <div className="month-header" style={{backgroundColor: monthColors[month-1]}}>
                            <h2>{monthNames[month-1]}</h2>
                        </div>

                        {
                            Object.keys(searchResults? searchResults[month] : calendarData[month]).map((day, j)=>{
                                return(
                                <div key={j}>
                                <div className="day-constainer" >
                                    <h2 style={{color: monthColors[month-1]}}>{day}</h2>
                                    <div className="events-container">
                                        {
                                            (searchResults? searchResults[month][day] : calendarData[month][day]).map((event, k)=>{
                                                return(
                                                    <div className="event-container" id={`event-${month}-${day}-${k}`} key={k} onClick={()=> eventEdit(`event-${month}-${day}-${k}`, event)}>
                                                        <p>{`${event.personName} ${event.AnniversaryType} ${event.type} ${getAge(event.date)}`}</p>
                                                    </div>
                                                )
                                            })
                                        }
                                    </div>
                                </div>

                                    <hr className="divition"/>
                            
                                
                                </div>
                                )
                            })
                        }
                        
                    </div>
                   )
               }) : loadingData  ? (<svg xmlns="http://www.w3.org/2000/svg" className="loading-wheel claendar-wheel" 
               viewBox="0 0 117.49 124.88" width="100" height="100"><defs></defs>
               <title>loading</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1">
                <path 
                 d="M78.84,18.33a49.61,49.61,0,0,0-28.75-2.12A50.63,50.63,0,0,0,24.57,30.68,51.5,51.5,0,0,0,11.19,57.34a52.14,
                 52.14,0,0,0,3.75,30.08,53.12,53.12,0,0,0,20.17,23.31A54.11,54.11,0,0,0,65.32,119a55.21,55.21,0,0,0,50-35.61,
                 1.12,1.12,0,0,1,1.45-.65,1.14,1.14,0,0,1,.67,1.4,58.76,58.76,0,0,1-19.55,28A59.79,59.79,0,0,1,65.63,124.7a60.64,
                 60.64,0,0,1-34.29-7.6A61.34,61.34,0,0,1,6.71,91.36,62.19,62.19,0,0,1,.46,55.78,63.25,63.25,0,0,1,15.23,22.27,
                 64.12,64.12,0,0,1,46.56,2.37a64.82,64.82,0,0,1,37.61.88l.08,0a8,8,0,0,1-5,15.2Z"/></g></g></svg>) 
                 : (
                     <div className="any-container">
                     <div className="any">
                     <h2>You don't have any event</h2>
                     <p>add one pressing </p>
                     <img src="img/add.svg" alt="add"/>
                     </div>
                    </div>)
             }
            
        </div>
    )
}
