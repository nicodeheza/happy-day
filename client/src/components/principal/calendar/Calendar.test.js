import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer,{act} from 'react-test-renderer';
import Calendar from './Calendar';
import {principalContext} from '../Principal';
import data from './data';


describe("<Calendar/>",()=>{
    global.fetch= jest.fn();
    const setShowCard= jest.fn();
    const setUpdateCalendar= jest.fn();
    const setEdit= jest.fn();
    beforeEach(()=>{
        fetch.mockClear();
        setShowCard.mockClear();
        setUpdateCalendar.mockClear();
        setEdit.mockClear();
    });

    it("if not events render message", async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve({})});
        });
        const calendar= render(
            <principalContext.Provider value={{
                updateCalendar: false,
                setUpdateCalendar,
                setEdit,
                searchFilters: {}
            }}>
                <Calendar setShowCard={setShowCard} showCard={'none'}/>
            </principalContext.Provider>
        );

        const message= await waitFor(()=>calendar.findByText("You don't have any event")); 
        expect(message).toBeInTheDocument();
    });
    it("if have events render events",async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve(data)});
        });
        let component
        await act(async()=>{
        component= renderer
        .create(   <principalContext.Provider value={{
            updateCalendar: false,
            setUpdateCalendar,
            setEdit,
            searchFilters: {}
        }}>
            <Calendar setShowCard={setShowCard} showCard={'none'}/>
        </principalContext.Provider>);

        });
        
        const tree= component.toJSON();
        expect(tree).toMatchSnapshot();

    });


    it("edit when click in an event", async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve(data)});
        });
        const calendar= render(
            <principalContext.Provider value={{
                updateCalendar: false,
                setUpdateCalendar,
                setEdit,
                searchFilters: {}
            }}>
                <Calendar setShowCard={setShowCard} showCard={'none'}/>
            </principalContext.Provider>
        );
       
        const event= await calendar.findByText(new RegExp('lololo'));
        fireEvent.click(event.parentNode);

        expect(event.parentNode).toHaveStyle('background-color: white')

        expect(setEdit).toBeCalledTimes(1);
        expect(setEdit).toBeCalledWith(data[1][1][0]);
        expect(setShowCard).toBeCalledWith('edit');

    });
    it("search by name",async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve(data)});
        });
        const calendar= render(
            <principalContext.Provider value={{
                updateCalendar: false,
                setUpdateCalendar,
                setEdit,
                searchFilters: {
                    from:'',
                    to:'',
                    name: 'abril',
                    type: 'any'
                }
            }}>
                <Calendar setShowCard={setShowCard} showCard={'none'}/>
            </principalContext.Provider>
        );

        const targetEvent= await calendar.findByText(new RegExp('abril'));
        expect(targetEvent).toBeInTheDocument();

        expect(calendar.container.querySelectorAll('p')).toHaveLength(1);

    });
    it("search by date", async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve(data)});
        });
        const calendar= render(
            <principalContext.Provider value={{
                updateCalendar: false,
                setUpdateCalendar,
                setEdit,
                searchFilters: {
                    from:'05/07',
                    to:'10/25',
                    name: '',
                    type: 'any'
                }
            }}>
                <Calendar setShowCard={setShowCard} showCard={'none'}/>
            </principalContext.Provider>
        );


        const may= await calendar.findByText('May');
        expect(may).toBeInTheDocument();

        const oct= await calendar.findByText('October');
        expect(oct).toBeInTheDocument();
        calendar.getByText('July');
        calendar.getByText('August');
        calendar.getByText('September');

        expect(calendar.queryByText('April')).toBeNull();
        expect(calendar.queryByText('March')).toBeNull();
        expect(calendar.queryByText('February')).toBeNull();
        expect(calendar.queryByText('January')).toBeNull();
        expect(calendar.queryByText('November')).toBeNull();
        expect(calendar.queryByText('December')).toBeNull();
        
        //console.log(prettyDOM(oct.parentElement.parentElement));
        expect(may.parentElement.parentElement).toContainHTML('<h2 style="color: rgb(105, 213, 3);">7</h2>');
        expect(may.parentElement.parentElement).not.toContainHTML('<h2 style="color: rgb(105, 213, 3);">6</h2>');
        expect(oct.parentElement.parentElement).toContainHTML('<h2 style="color: rgb(253, 97, 38);">25</h2>');
        expect(oct.parentElement.parentElement).not.toContainHTML('<h2 style="color: rgb(253, 97, 38);">30</h2>');

    });
    it("search by type", async()=>{
        fetch.mockImplementation(()=>{
            return Promise.resolve({json: () => Promise.resolve(data)});
        });
        const calendar= render(
            <principalContext.Provider value={{
                updateCalendar: false,
                setUpdateCalendar,
                setEdit,
                searchFilters: {
                    from:'',
                    to:'',
                    name: '',
                    type: 'anniversary'
                }
            }}>
                <Calendar setShowCard={setShowCard} showCard={'none'}/>
            </principalContext.Provider>
        );

        await calendar.findAllByText(new RegExp('anniversary', 'i'));
        expect(calendar.queryByText(new RegExp('birthday','i'))).toBeNull();
    });
});