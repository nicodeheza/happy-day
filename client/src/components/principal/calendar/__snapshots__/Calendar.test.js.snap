// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar/> if have events render events 1`] = `
<div
  className="calendar"
>
  <div
    className="month-container"
    id="month1"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#216b8e",
        }
      }
    >
      <h2>
        January
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#216b8e",
            }
          }
        >
          1
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-1-1-0"
            onClick={[Function]}
          >
            <p>
              lololo  birthday (31 eyars)
            </p>
          </div>
          <div
            className="event-container"
            id="event-1-1-1"
            onClick={[Function]}
          >
            <p>
              lo  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#216b8e",
            }
          }
        >
          22
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-1-22-0"
            onClick={[Function]}
          >
            <p>
              roco  birthday (55 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month2"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#27a3a3",
        }
      }
    >
      <h2>
        February
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#27a3a3",
            }
          }
        >
          7
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-2-7-0"
            onClick={[Function]}
          >
            <p>
              bobo  birthday (23 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#27a3a3",
            }
          }
        >
          8
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-2-8-0"
            onClick={[Function]}
          >
            <p>
              jojo  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month3"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#149e84",
        }
      }
    >
      <h2>
        March
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#149e84",
            }
          }
        >
          7
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-3-7-0"
            onClick={[Function]}
          >
            <p>
              mar  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#149e84",
            }
          }
        >
          8
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-3-8-0"
            onClick={[Function]}
          >
            <p>
              yoyo  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month4"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#00ae4d",
        }
      }
    >
      <h2>
        April
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#00ae4d",
            }
          }
        >
          5
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-4-5-0"
            onClick={[Function]}
          >
            <p>
              abril  birthday (40 eyars)
            </p>
          </div>
          <div
            className="event-container"
            id="event-4-5-1"
            onClick={[Function]}
          >
            <p>
              test  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#00ae4d",
            }
          }
        >
          6
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-4-6-0"
            onClick={[Function]}
          >
            <p>
              juna  birthday (22 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month5"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#69d503",
        }
      }
    >
      <h2>
        May
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#69d503",
            }
          }
        >
          6
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-5-6-0"
            onClick={[Function]}
          >
            <p>
              pepito move anniversary 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#69d503",
            }
          }
        >
          7
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-5-7-0"
            onClick={[Function]}
          >
            <p>
              pop  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#69d503",
            }
          }
        >
          10
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-5-10-0"
            onClick={[Function]}
          >
            <p>
              prueba  birthday 
            </p>
          </div>
          <div
            className="event-container"
            id="event-5-10-1"
            onClick={[Function]}
          >
            <p>
              test  birthday 
            </p>
          </div>
          <div
            className="event-container"
            id="event-5-10-2"
            onClick={[Function]}
          >
            <p>
              jose  birthday (31 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#69d503",
            }
          }
        >
          11
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-5-11-0"
            onClick={[Function]}
          >
            <p>
              jhon and eli casamiento  anniversary (41 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#69d503",
            }
          }
        >
          25
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-5-25-0"
            onClick={[Function]}
          >
            <p>
              may  birthday (12 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month7"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#edc01c",
        }
      }
    >
      <h2>
        July
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#edc01c",
            }
          }
        >
          5
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-7-5-0"
            onClick={[Function]}
          >
            <p>
              juju  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#edc01c",
            }
          }
        >
          6
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-7-6-0"
            onClick={[Function]}
          >
            <p>
              juan y ana  casamiento  anniversary (22 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#edc01c",
            }
          }
        >
          25
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-7-25-0"
            onClick={[Function]}
          >
            <p>
              peter  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month8"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#faa818",
        }
      }
    >
      <h2>
        August
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#faa818",
            }
          }
        >
          8
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-8-8-0"
            onClick={[Function]}
          >
            <p>
              agus  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#faa818",
            }
          }
        >
          25
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-8-25-0"
            onClick={[Function]}
          >
            <p>
              aggg  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month9"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#f19616",
        }
      }
    >
      <h2>
        September
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#f19616",
            }
          }
        >
          15
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-9-15-0"
            onClick={[Function]}
          >
            <p>
              hugo  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#f19616",
            }
          }
        >
          17
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-9-17-0"
            onClick={[Function]}
          >
            <p>
              sep  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#f19616",
            }
          }
        >
          20
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-9-20-0"
            onClick={[Function]}
          >
            <p>
              lol  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month10"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#fd6126",
        }
      }
    >
      <h2>
        October
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#fd6126",
            }
          }
        >
          25
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-10-25-0"
            onClick={[Function]}
          >
            <p>
              jhon doe  birthday (34 eyars)
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#fd6126",
            }
          }
        >
          30
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-10-30-0"
            onClick={[Function]}
          >
            <p>
              pepo  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month11"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#9458a9",
        }
      }
    >
      <h2>
        November
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#9458a9",
            }
          }
        >
          5
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-11-5-0"
            onClick={[Function]}
          >
            <p>
              nono  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#9458a9",
            }
          }
        >
          22
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-11-22-0"
            onClick={[Function]}
          >
            <p>
              pepe  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
  <div
    className="month-container"
    id="month12"
  >
    <div
      className="month-header"
      style={
        Object {
          "backgroundColor": "#6258a2",
        }
      }
    >
      <h2>
        December
      </h2>
    </div>
    <div>
      <div
        className="day-constainer"
      >
        <h2
          style={
            Object {
              "color": "#6258a2",
            }
          }
        >
          12
        </h2>
        <div
          className="events-container"
        >
          <div
            className="event-container"
            id="event-12-12-0"
            onClick={[Function]}
          >
            <p>
              tato  birthday 
            </p>
          </div>
        </div>
      </div>
      <hr
        className="divition"
      />
    </div>
  </div>
</div>
`;
